python/ql/src/semmle/python/regex.qll

```diff
@@ -124,16 +124,48 @@ abstract class RegexString extends Expr {
         )
     }
 
+    /** Escaped characters without any special handling (yet) */
+    private predicate singleEscape(int i) {
+        exists(string c |
+          c = this.getChar(i) and
+          c != "x" and c != "u" and c != "U" and c != "N"
```

Comment:
This might benefit from the use of set literals: `not c in ["x", "u", "U", "N"]`.
(At which point you can dispense with `c` and the `exists` entirely.)

Reply:
Ah, excellent :)